packages/dara-components/dara/components/graphs/definitions.py

```diff
@@ -31,6 +32,101 @@ class EditorMode(str, Enum):
     """Resolver mode - allows users to accept edges"""
 
 
+class ArrowType(str, Enum):
+    NONE = 'none'
+    NORMAL = 'normal'
+    FILLED = 'filled'
+    EMPTY = 'empty'
+    SOFT = 'soft'
+
+
+class CenterSymbol(str, Enum):
+    NONE = 'none'
+    CROSS = 'cross'
+    QUESTION = 'question'
+    BIDIRECTED = 'bidirected'
+
+
+class GraphLegendType(str, Enum):
+    SPACER = 'spacer'
+    EDGE = 'edge'
+    NODE = 'node'
+
+
+class GraphLegend(BaseModel):
```

Comment:
I think it's more natural in Python to be represent the union as subclasses, maybe with a base class.

Or, just use a union of typed dicts can suffice:


```py
# ...
class SpacerLegend(TypedDict):
  type: Literal['spacer']
  # ... other fields


GraphLegend = Union[SpacerLegend, EdgeLegend, NodeLegend]
```

Probably easiest to use, doesn't require extra imports. Also the enum for legend type is unnecessary

Reply:
Addressed in person as well